scripts/generateDeepExports.mjs

```diff
@@ -0,0 +1,362 @@
+  console.warn(
+    `✓ Created deep export for ${componentName} in ${path.relative(process.cwd(), targetPackagePath)}`,
+  );
+}
+
+/**
+ * Process a base package and create deep exports for its components in target packages.
+ * @param {string} basePackageName
+ */
+async function processPackage(basePackageName) {
+  const basePackagePath = path.join(process.cwd(), 'packages', basePackageName);
+  /** @type {{targets: string[]; ignore: string[]; utilities: string[];}} */
+  const targetPackages = PACKAGE_MAPPINGS[basePackageName] || {};
+
+  if (!existsSync(basePackagePath)) {
+    console.warn(`Base package not found: ${basePackagePath}`);
+    return;
+  }
+
+  console.warn(`\n📦 Processing ${basePackageName} -> ${targetPackages.targets.join(' -> ')}...`);
+
+  // Get component directories from the base package
+  const componentDirectories = [
+    await getComponentDirectories(basePackagePath, targetPackages.ignore),
+  ];
+  const sourcePackageChain = [basePackageName];
+
+  for (const targetPackage of targetPackages.targets) {
+    const targetPackagePath = path.join(process.cwd(), 'packages', targetPackage);
+
+    if (!existsSync(targetPackagePath)) {
+      console.warn(`Target package not found: ${targetPackagePath}`);
+      return;
+    }
+
+    // Get additional components specific to pro/premium
+    // eslint-disable-next-line no-await-in-loop
+    const targetComponentDirs = await getComponentDirectories(
+      targetPackagePath,
+      targetPackages.ignore,
+    );
+    const targetDirectories = componentDirectories.map((dirs) =>
+      dirs.filter((name) => !targetComponentDirs.includes(name)),
+    );
+
+    console.warn(`\nCreating deep exports for ${targetDirectories.flat().length} components`);
+
+    const componentTasks = targetDirectories.flatMap((componentNames, i) =>
+      componentNames.map((componentName) =>
+        createDeepExportFile(
+          targetPackagePath,
+          componentName,
+          sourcePackageChain[i],
+          targetPackages.utilities,
+          sourcePackageChain,
+        ),
+      ),
+    );
+
+    const utilityTasks = targetPackages.utilities
+      .filter((utilityName) => {
+        const utilityDirPath = path.join(targetPackagePath, 'src', utilityName);
+        return existsSync(utilityDirPath);
+      })
+      .map((utilityName) =>
+        createDeepExportFile(
+          targetPackagePath,
+          utilityName,
+          sourcePackageChain[0],
+          targetPackages.utilities,
+          sourcePackageChain,
+        ),
+      );
+
+    // eslint-disable-next-line no-await-in-loop
+    await Promise.all([...componentTasks, ...utilityTasks]);
+
+    // Merge component directories for the next target package
+    componentDirectories.push(targetComponentDirs);
+    sourcePackageChain.push(targetPackage);
+  }
+}
+
+// Main execution
+async function main() {
+  console.warn('🚀 Generating deep exports for pro and premium packages...\n');
```

Comment:
Shouldn't all these informational logs be `console.log` instead of `console.warn`?